### PR TITLE
Provisional Assessment Report - filters

### PR DESCRIPTION
**Pull request by colinbruce**
#### What
Add filters to Provisional Assessment Report

#### Ticket
[CBO311 refine provisional assessment](https://dsdmoj.atlassian.net/browse/CBO-311)

#### Why
The business need to drill into the data further than the original, aggregated, report allowed

#### How
Create a new output that the users can create, for a given date range, that allows them to perform Transactional Profiling

--------

#### TODO (wip)

 - [X] implement new API, MI report
 - [X] allow calling of new report from existing MI page
 - [x] improve test coverage
 - [x] refactor JS
 - [ ] implement re-design of MI page - on hold, will be separate ticket

### CODE REVIEW

File: app/assets/javascripts/modules/case_worker/admin/management_information.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  },
+
+  bindEvents: function () {
+    this.provisionalAssessmentsDateFieldChange();
+    this.provisionalAssessmentsDownloadClicked();
+  },
+
+  buildDataArray: function() {
+    var data = {};
+    data.api_key='64c9177b-e5a0-4ac5-9f2b-33166597c436';
```
Comment (jsugarman) [inline app/assets/javascripts/modules/case_worker/admin/management_information.js]: is this key ok here?

File: app/services/reports/provisional_assessments_dates.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+module Reports
+  class ProvisionalAssessmentsDates
```
Comment (jsugarman) [inline app/services/reports/provisional_assessments_dates.rb]: not a blocker  - `ProvisionalAssessmentsByDate[s]` as a name may be clearer

**Review by jsugarman — COMMENTED** [review]
Looks good. just want to confirm if the API key needs to be an env var or other kind of secret?

File: app/assets/javascripts/modules/case_worker/admin/management_information.js (hunk 1 above)
Comment (colinbruce) [reply app/assets/javascripts/modules/case_worker/admin/management_information.js]: Good catch... that should have been current_user.api_key... I wonder where that went! 😨

**Review by jsugarman — APPROVED** [review]
LGTM 👍

**Pull request merged**